### ISSUE TITLE
we need class diagrams 

### ISSUE THREAD

**Issue by clear-m** [danielgindi/Charts#1299]
Hi there!

Your library is so huge! I think that any newcomers really appreciate some help with inheritance chains overview.
Please provide some kind of class diagrams in future - it will be very helpful !

Thanks for made this lib available for us )

**Comment by liuxuan30** [comment]
well.. it won't take you much time to trace down. The name already hints a lot

**Comment by clear-m** [comment]
Yeah - that's what I already did - but it can be even more obvious with diagrams - it could be done like in here - https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/Documentation/CocoaLumberjackClassDiagram.png

**Comment by danielgindi** [comment]
This could be great - and will definitely be better to do based on v3 - where we changed some of the structure, made some things more protocol-oriented etc.

**Comment by liuxuan30** [comment]
@danielgindi actually I have been testing jazzy these days to generate a good looking doc, probably can help, however, jazzy has some bugs and issues pending reply :( basically it would work, but not in a neat way

**Comment by clear-m** [comment]
Hi there!
I wish to share with you simple but yet representative sample of current charts 3.0 class diagrams generated with a help of [swift-relationship-graph tool](https://github.com/Sumolari/swift-relationship-graph)

[charts_3.0_beta.pdf](https://github.com/danielgindi/Charts/files/538755/charts_3.0_beta.pdf)

I hope it would be helpful for someone )
